DataRepo/utils/file_utils.py

```
@@ -0,0 +1,287 @@
+    sheet_name = sheet
+    sheets = pd.ExcelFile(filepath, engine="openpyxl").sheet_names
+    if str(sheet_name) not in sheets:
+        sheet_name = 0
+
+    # Note, setting `mangle_dupe_cols=False` would overwrite duplicates instead of raise an exception, so we're
+    # checking for duplicate headers manually here.
+    return (
+        pd.read_excel(
+            filepath,
+            nrows=1,  # Read only the first row
+            header=None,
+            sheet_name=sheet_name,  # The first sheet
+            engine="openpyxl",
+        )
+        .squeeze("columns")
+        .iloc[0]
+    )
+
+
+def _read_headers_from_tsv(filepath):
+    # Note, setting `mangle_dupe_cols=False` would overwrite duplicates instead of raise an exception, so we're
+    # checking for duplicate headers manually here.
+    return (
+        pd.read_table(
+            filepath,
+            nrows=1,
+            header=None,
+        )
+        .squeeze("columns")
+        .iloc[0]
+        .to_list()
+    )
+
+
+def _read_headers_from_csv(filepath):
+    # Note, setting `mangle_dupe_cols=False` would overwrite duplicates instead of raise an exception, so we're
+    # checking for duplicate headers manually here.
+    return (
+        pd.read_csv(
+            filepath,
+            nrows=1,
+            header=None,
+        )
+        .squeeze("columns")
+        .iloc[0]
+        .to_list()
+    )
+
+
+def headers_are_as_expected(expected, headers):
+    """Confirms all headers are present, irrespective of case and order."""
+    return sorted([s.lower() for s in headers]) == sorted([s.lower() for s in expected])
+
+
+def get_sheet_names(filepath):
+    """
+    Returns a list of sheet names in an excel file.  Returns None if the file is not an excel file.
+    """
+    try:
+        return pd.ExcelFile(filepath, engine="openpyxl").sheet_names
+    except (InvalidFileException, ValueError, BadZipFile):  # type: ignore
+        return None  # Not an excel file
```

C. Might be better to simply leave this off and let the exception be raised.

---

I'll think about this.  I think that I call this without gating it via a call to `is_excel`. If I rework that logic, then raising would make sense.

---

Implemented on `input_loader_means2` branch.